packages/camera/tizen/src/orientation_manager.cc

```diff
@@ -0,0 +1,154 @@
+// Copyright 2021 Samsung Electronics Co., Ltd. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "orientation_manager.h"
+
+#include <flutter/encodable_value.h>
+
+#include <string>
+
+#include "device_method_channel.h"
+#include "log.h"
+
+bool OrientationTypeToString(OrientationType orientation_type,
+                             std::string& orientation) {
+  switch (orientation_type) {
+    case OrientationType::kPortraitUp:
+      orientation = "portraitUp";
+      return true;
+    case OrientationType::kLandscapeLeft:
+      orientation = "landscapeLeft";
+      return true;
+    case OrientationType::kPortraitDown:
+      orientation = "portraitDown";
+      return true;
+    case OrientationType::kLandscapeRight:
+      orientation = "landscapeRight";
+      return true;
+    default:
+      LOG_WARN("Unknown OrientationType!");
+      return false;
+  }
+}
+
+bool StringToOrientationType(std::string orientation,
+                             OrientationType& orientation_type) {
+  if (orientation == "portraitUp") {
+    orientation_type = OrientationType::kPortraitUp;
+    return true;
+  } else if (orientation == "landscapeLeft") {
+    orientation_type = OrientationType::kLandscapeLeft;
+    return true;
+  } else if (orientation == "portraitDown") {
+    orientation_type = OrientationType::kPortraitDown;
+    return true;
+  } else if (orientation == "landscapeRight") {
+    orientation_type = OrientationType::kLandscapeRight;
+    return true;
+  }
+  LOG_WARN("Unknown OrientationType!");
+  return false;
+}
+
+OrientationManager::OrientationManager(
+    DeviceMethodChannel* device_method_channel,
+    OrientationType lens_orientation, bool is_front_lens_facing)
+    : device_method_channel_(device_method_channel),
+      lens_orientation_(lens_orientation),
+      is_front_lens_facing_(is_front_lens_facing) {
+  LOG_DEBUG("is_front_lens_facing_[%s]",
+            is_front_lens_facing_ ? "true" : "false");
+
+  // Send initial orientation
+  last_device_orientation_ = GetDeviceOrientationType();
+  target_orientation_ = ConvertOrientation(last_device_orientation_);
+  SendOrientation(target_orientation_);
+}
+
+OrientationManager::~OrientationManager() {}
+
+OrientationType OrientationManager::ConvertOrientation(
+    OrientationType orientation_event_type, bool to_target /* = true */) {
+  int degree = static_cast<int>(orientation_event_type);
+  if (is_front_lens_facing_) {
+    degree = 180 + degree;
+  }
+  int target =
+      (degree + (static_cast<int>(lens_orientation_) * (to_target ? 1 : -1))) %
+      360;
```

Comment:
Why adding 180 degrees if `is_front_lens_facing_`?

What does `to_target` mean in this function?

Reply:
> Why adding 180 degrees if `is_front_lens_facing_`?

This is because when using the front camera, plugin want it to work in mirror mode.

> What does to_target mean in this function?

There is a case that converts an orientation that already converted and sent from the native area when received it from ```dart```, Please refer to ```CameraDevice :: LockCaptureOrientation```.